### PR TITLE
Expand CI coverage

### PR DESCRIPTION
**Pull request by isovector**
Now that #29 is finished, let's see which versions of GHC this thing compiles with.

### CODE REVIEW

**Comment by jkachmar** [comment]
Is there anything I can do to help with this? Looks like it's diverged a bit since my last PR.

**Comment by isovector** [comment]
@jkachmar we're waiting on getting #67 merged first, which should solve all of the TH woes. Thanks for staying keen about it though!

File: src/Polysemy/Writer.hs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE BlockArguments  #-}
+{-# LANGUAGE CPP             #-}
```
Comment (isovector) [inline src/Polysemy/Writer.hs]: why

**Comment by isovector** [comment]
okay! so core polysemy works! But i'm not sure how to get cabal to build polysemy-plugin against the polysemy that exists in the repo

**Comment by jkachmar** [comment]
> But i'm not sure how to get cabal to build `polysemy-plugin` against the `polysemy` that exists in the repo

#123 should address this; you can tell `cabal-install` about projects that are in the same directory by setting a `cabal.project` file with some of this information called out there.

**Comment by ocharles** [comment]
"Mute this thread"

**Comment by jhenahan** [comment]
This is my favorite repo on all of GitHub.

**Pull request merged**